File: apps/framework-cli/tests/ts/simple.ts
type Key<T extends string | number> = T;

enum StringedEnum {
  A = "TEST",
}

interface MyModel {
  name: Key<string>;
  age: number;
  // abc: MyEnum;
  flag: boolean;
  "test-key": string;
  arr: string[];
  opt?: string;
  enum3: StringedEnum;
}

File: apps/framework-cli/tests/python/simple.py
from dataclasses import dataclass, field
from enum import Enum
from typing import List, Optional


type Key[T: (str, int)] = T 


# Might be a solid idea to use the field specifier parameters to rename errand fields?
# https://typing.readthedocs.io/en/latest/spec/dataclasses.html#field-specifier-parameters

class Status(Enum):
    OK = "ok"
    ERROR = "error"


@dataclass 
class MyModel:
    name: Key[str]
    age: int
    flag: bool
    status: Status
    test_key: str = field(alias="test-key")
    arr: List[str]
    opt: Optional[str]

